### ISSUE TITLE
Barbarian Levels

### ISSUE THREAD

**Issue by GeraldTo** [5e-bits/5e-database#366]
under  classes/barbarian/levels there are 2 level 6 objects. its not an exact duplicate though since the 1st one has information that the other is missing and vice versa. I believe the correct object should be:
```json
{
		"level": 6,
		"ability_score_bonuses": 1,
		"prof_bonus": 3,
		"feature_choices": [],
		"features": [
			{
				"index": "mindless-rage",
				"name": "Mindless Rage",
				"url": "/api/features/mindless-rage"
			}
		],
		"class_specific": {
			"creating_spell_slots": [],
			"rage_count": 4,
			"rage_damage_bonus": 2,
			"brutal_critical_dice": 0
		},
		"index": "barbarian-6",
		"class": {
			"index": "barbarian",
			"name": "Barbarian",
			"url": "/api/classes/barbarian"
		},
		"subclass": {
			"index": "berserker",
			"name": "Berserker",
			"url": "/api/subclasses/berserker"
		},
		"url": "/api/classes/barbarian/levels/6"
},
```

**Comment by bagelbits** [comment]
Why are we combining the Berserker level 6 with the Barbarian level 6?

**Comment by GeraldTo** [comment]
I just misunderstood the way the subclass levels and levels are supposed to be structured.  Wouldn't it make more sense to have separate lists though? Just an idea. Or maybe have a label on top of the index for the class or subclass?

**Comment by fergcb** [comment]
> Wouldn't it make more sense to have separate lists though?

Not really, imo. Subclass levels are identical in structure to class levels, but for the subclass id, and are almost always gonna be processed the same way by clients.

*Maybe* there's a case to be made about having the `/classes/{index}/levels` endpoint return only levels belonging to the base class? Or perhaps having a `subclass` query parameter to filter subclass levels?

**Comment by bagelbits** [comment]
`/subclasses/{index}/levels/` and `/subclasses/{index}/levels/{level}` already exist. I could also see an argument for `classes/{index}/levels` returning only the base class.

**Comment by fergcb** [comment]
I'm aware that they exist. I was just offering an alternative given the suggestion that having all the levels (subclass or not) in the `/classes/{index}/levels` might be confusing.

My suggestion of a *query parameter* on that endpoint would allow consumers to query levels for a base class *and* a particular subclass. An absence of that parameter would just return the base levels.

E.g. `/classes/barbarian/levels?subclass=beserker` could return all the base barbarian levels *and* the barbarian>berserker levels.

I hope that makes more sense?

**Comment by bagelbits** [comment]
Oh yes! That makes more sense. Thank you for the clarification. I really like that!

**Comment by bagelbits** [comment]
I can 100% see the value of reworking the endpoint to do that and I'm here for it. :D

**Comment by bagelbits** [comment]
Oh. This is an issue for the API and not the database I believe. I'll migrate it over.
